rules/repositories.bzl

```diff
@@ -58,20 +58,52 @@ def _get_bazel_version():
     return struct(major = 0, minor = 0, patch = 0)
 
 def rules_ios_dependencies(
-        load_xchammer_dependencies = False):
-    """Fetches repositories that are dependencies of `rules_ios`.
+        load_xchammer_dependencies = False,
+        load_bzlmod_dependencies = True):
+    """Fetches repositories that are public dependencies of `rules_ios`.
 
     Args:
         load_xchammer_dependencies: if `True` loads XCHammer and xcbuildkit dependencies (this is considered "alpha" software at the moment, see `README`)
+        load_bzlmod_dependencies: if `True` loads dependencies that are available via bzlmod (set to True when using WORKSPACE, and False when using bzlmod)
     """
-    bazel_version = _get_bazel_version()
-    if bazel_version.major == "5":
-        # Minimum commit vetted of rules_swift - not necessarily exclusive with
-        # this bazel version or rules_apple version but tested on CI as so.
+
+    if load_bzlmod_dependencies:
+        _rules_ios_bzlmod_dependencies()
+
+    if load_xchammer_dependencies:
+        _rules_ios_xchammer_dependencies()
+
+    # This goes after xchammer because it also depends on XcodeGen repo.
+    # If legacy project generator is deprecated this can be removed.
+    _rules_ios_legacy_xcodeproj_dependencies()
+
+def rules_ios_dev_dependencies(
```

Comment:
Why do we need this `rules_ios_dev_dependencies` function - this just adds so much complexity to the setup and repo. Bazel should only download what it is using unless you start running wildcard queries which is a performance problem that this looks like you are attempting to fix

Reply:
Because in bzlmod you mark things as being development only dependencies when they aren't required by consumers of the rules. No one consuming rules_ios needs to download buildifier, which bzlmod solves. 

This specific change lets us reuse `repositories.bzl` while we transition to bzlmod completely (Bazel 9 I believe).

Reply:
Some examples: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/main/MODULE.bazel#L30

Reply:
> this just adds so much complexity to the setup

fwiw this shouldn't need to be run by end users, it's just a way for us internally to fetch development repos in both bzlmod / non-bzlmod workflows.